master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java

```diff
@@ -344,19 +484,26 @@ static List<WorkerInfo> generateRackAwareWorkers(List<WorkerInfo> workers) {
   private static List<Integer> roundRobin(
       Map<WorkerInfo, Tuple2<List<PartitionLocation>, List<PartitionLocation>>> slots,
       List<Integer> partitionIds,
-      List<WorkerInfo> workers,
+      List<WorkerInfo> primaryWorkers,
+      List<WorkerInfo> replicaWorkers,
       Map<WorkerInfo, List<UsableDiskInfo>> slotsRestrictions,
       boolean shouldReplicate,
       boolean shouldRackAware,
-      int availableStorageTypes) {
+      int availableStorageTypes,
+      boolean replicaSameAsPrimary) {
```

Comment:
Given the number of parameters have increased, and some of them are not obvious from first read - can you update the javadoc as well ?